ui: statements page

vladlos opened the pull request:
statements page redesign, because of table component is reused on other pages, it changed view there also.

Resolves: #42154 

Release note (ui): none

![Screenshot 2020-01-10 at 17 56 32](https://user-images.githubusercontent.com/12850886/72166868-ba4d5380-33d2-11ea-84b4-c8cd681d5817.png)
![Screenshot 2020-01-10 at 17 56 40](https://user-images.githubusercontent.com/12850886/72166872-bd484400-33d2-11ea-934a-01ed55e27861.png)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/43875)
<!-- Reviewable:end -->

vilterp commented:
This may be an @Annebirzin question, but why put the logical plan and stats on different tabs? Just seems like more clicks to get the same information.

Annebirzin commented:
@vilterp the main thinking there was for scalability. It seemed that the statement schema or logical plans could get very long in which case showing everything in one view would require lots of scrolling (not to mention the logical plan might get lost below the fold).

But if very long statement schemas or logical plans are an edge case then we could consider keeping them in one tab. We can also test this version out with users to see if they are able to access logical plans intuitively.

dhartunian reviewed (changes requested):
> Reviewed 14 of 36 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43875)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster)

___
*[pkg/ui/src/views/shared/components/summaryCard/imdex.tsx, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_GsmoD7EmAvY0Ec6R:-Ly_GsmoD7EmAvY0Ec6S:b-nm72fz) ([raw file](https://github.com/cockroachdb/cockroach/blob/827e768a309a1aa3942326d3bc4711426f3464d4/pkg/ui/src/views/shared/components/summaryCard/imdex.tsx#L1)):*
> ```XML
> // Copyright 2018 The Cockroach Authors.
> ```

@elkmaster Please rename this file to `index.tsx` and change the license to match the other javascript files.


<!-- Sent from Reviewable.io -->

dhartunian reviewed (changes requested):
As discussed offline with @elkmaster, we will be breaking this PR up into smaller pieces.

> Reviewed 2 of 36 files at r1, 7 of 7 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-:-Lya-S-W7OSXM1xhxLgG:bs7to8t)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster)

___
*[pkg/ui/src/views/app/components/empty/index.tsx, line 34 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_O9sQ0zEFzoSx0x_4:-Ly_O9sQ0zEFzoSx0x_5:bkt485f) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/app/components/empty/index.tsx#L34)):*
> ```XML
>       <h2 className="empty-container__title">{title}</h2>
>       <p className="empty-container__description">{description}</p>
>       <Button className="empty-container__button" type="primary">{button}</Button>
> ```

This button doesn't link to anything. Am I misunderstanding its purpose?
___
*[pkg/ui/src/views/app/components/empty/styles.styl, line 48 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_Zsgc8R3scwRnuIwh:-Ly_Zsgc8R3scwRnuIwi:b-8h9vim) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/app/components/empty/styles.styl#L48)):*
> ```Stylus
>       background-color $green
>       opacity 0.7
>       color $adminui-white !important
> ```

we should avoid !important. Why is this necessary?
___
*[pkg/ui/src/views/app/components/Search/index.tsx, line 11 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly__EO12FkWJBLXdaiJ:-Ly__EO12FkWJBLXdaiK:b-brrcms) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/app/components/Search/index.tsx#L11)):*
> ```XML
> // licenses/APL.txt.
> 
> import { Button, Form, Icon, Input } from "antd";
> ```

Icon is no longer used, please remove from the imports
___
*[pkg/ui/src/views/shared/components/sql/box.tsx, line 24 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_ZBQC3EIqy3SUJ8N6:-Ly_ZBQD2Wz4uA4lTsEN:bmt2q04) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/shared/components/sql/box.tsx#L24)):*
> ```XML
> }
> 
> const regExpTables = /(?<=(FROM|INTO|TABLE|UPDATE|CREATE)\s)([^\s([]+)/g;
> ```

I'll look into whether the backend can tell you what the table name is.

FYI Lookbehind regexp syntax is not supported in Firefox yet, avoid using it for now.
https://github.com/sindresorhus/ip-regex/issues/24
___
*[pkg/ui/src/views/shared/components/summaryBar/index.tsx, line 61 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_oTN_BOh-uqewMfF1:-Ly_oTNa2vwAt910A62l:bfdtpdc) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/shared/components/summaryBar/index.tsx#L61)):*
> ```XML
> }
> 
> export function computeValue(value: number, format: (n: number) => string = numberToString) {
> ```

Since this function will be used outside this module, can you give it a more descriptive name? Maybe `formatNumberForDisplay`
___
*[pkg/ui/src/views/statements/barCharts.tsx, line 209 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_xJNI0mnm5nnLJk0X:-Ly_xJNO4Ei37_LhSJDo:bilwo1n) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/statements/barCharts.tsx#L209)):*
> ```XML
>   return {
>     firstAttemptsBarChart() {
>       return firstAttempts;
> ```

These values being returned are no longer bar charts so you either need to make a new file, or restructure the naming in this one. 


<!-- Sent from Reviewable.io -->

vladlos committed:
ui: assets

Added assets

Release note (ui): none

vladlos committed:
ui: statements empty

Added empty screen for statements and palette

Release note (ui): none

vladlos committed:
ui: search input

updated search input icons and styles

Release note (ui): none

vladlos committed:
ui: previous/next tooltip on pagination

removed previous/next tooltip on pagination

hide pagination if there is not result on the page in

updated statements styles

Release note (ui): none

vladlos committed:
ui: statements page

restyled statement components

Release note (ui): none

vladlos committed:
ui: statements details tabs

added tabs to statement details, created summary card, styled statements details tab overview.

Release note (ui): none

vladlos committed:
ui: statements logical plan

updated statement logical plan tab

Release note (ui): none

vladlos committed:
ui: statements details execution stats tab

restyled execution stats tab

Release note (ui): none

vladlos committed:
ui: statements details bar charts

renamed computeValue function, moved countBreakdown constants to statementDetails component

Release note (ui): none

vladlos committed:
ui: statements details sql box

Added tooltip when hovering over tables in statement schema to show full table shema and added link to table detail from statment schema

Release note (ui): none

vladlos committed:
ui: styles

updated styles and fixed lint

Release note (ui): none

vladlos committed:
ui: statement trigger module

Removed statement triggered model from bottom of the page when clicking a statement link.

Release note (ui): none

vladlos committed:
ui: statements no results

Added no results state to tables

Release note (ui): none

vladlos committed:
ui: tables font

Updated all fonts in tables except the description

Release note (ui): none

dhartunian reviewed (changes requested):
Thanks for restructuring this @elkmaster. Let me know if you have any questions about my comments.

elkmaster commented on pkg/ui/src/views/statements/barCharts.tsx:
```diff
@@ -193,57 +200,25 @@ export function countBreakdown(s: StatementStatistics) {
-    .range([0, 100]);
+  // const scale = d3.scale.linear()
+  //   .domain([0, count])
+  //   .range([0, 100]);
```
this change is outdated. in latest commit it's not commented.

elkmaster commented on pkg/ui/src/views/shared/components/sortabletable/index.tsx:
```diff
@@ -68,6 +68,7 @@ interface TableProps {
   // a full-width area below it when expanded.
   expandableConfig?: ExpandableConfig;
   drawer?: boolean;
+  firstCellHeader?: boolean;
```
this prop is needed to add border-right for first cell of the row like here:
![Снимок экрана 2020-01-28 в 10 25 15](https://user-images.githubusercontent.com/12850886/73249127-ecc1c380-41bc-11ea-8e50-886e94c75e3b.png)

I'll rename prop to be more clear

vladlos committed:
ui: sorttable firstCellBordered props

added firstCellBordered for highlight the first column in rows

Release note (ui): none

vladlos committed:
ui: sortable styles

removed unnecessary styles

Release note (ui): none

vladlos committed:
ui: pagination

used hideOnSinglePage option for hide/show pagination

Release note (ui): none

vladlos committed:
ui: sql box

Removed highlighting table name from sql box

Release note (ui): none

vladlos committed:
ui: empty network

added default props for empty

Release note (ui): none

dhartunian committed:
ui: fix linting errors

- Removed unnecessary wrapping in `mapDispatchToProps`
- Removed unused imports

Release note: None

Release note (<category, see below>): <what> <show> <why>

dhartunian commented:
bors r+
